dark/process.py

```diff
@@ -16,13 +18,19 @@ class Executor:
         of commands will be written to this descriptor.
     """
 
-    def __init__(self, dryRun=False, stdout=None, stderr=None):
-        self.dryRun = dryRun
+    def __init__(
+        self,
+        dryRun: bool = False,
+        stdout: Optional[Union[TextIO, StringIO]] = None,
+        stderr: Optional[Union[TextIO, StringIO]] = None,
+    ) -> None:
+        self._dryRun = dryRun
         self.stdout = stdout
         self.stderr = stderr
         self.log = [f"# Executor created at {ctime(time())}. Dry run = {dryRun}."]
 
-    def dryRun(self):
+    @property
+    def dryRun(self) -> bool:
```

Comment:
You say my properties are idiosyncratic... why do you keep this method as a property that just returns an attribute?

Reply:
I did it to keep backwards compatibility, in case someone was using the function. But as you pointed out, the function referred to a non-existent `self` variable. So I'm going to change it, thanks :-)